src/Traits/HasRoles.php

```diff
@@ -84,6 +84,59 @@ public function scopeRole(Builder $query, $roles): Builder
     }
 
     /**
+     * Scope the model query to certain permissions only.
+     *
+     * @param \Illuminate\Database\Eloquent\Builder $query
+     * @param string|array|\Spatie\Permission\Contracts\Permission|\Illuminate\Support\Collection $permissions
+     *
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function scopePermission(Builder $query, $permissions): Builder
+    {
+        if ($permissions instanceof Collection) {
```

Comment:
Let's put everything between line 96 and 108 in it's own dedicated protected function `convertToPermissionModels`